src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -198,25 +200,36 @@ struct PyModuleDef pythonmonkey =
   PythonMonkeyMethods
 };
 
+PyObject *SpiderMonkeyError = NULL;
+
 PyMODINIT_FUNC PyInit_pythonmonkey(void)
 {
   PyDateTime_IMPORT;
 
-  if (!JS_Init())
+  if (!JS_Init()) {
+    PyErr_SetString(SpiderMonkeyError, "Spidermonkey could not be initialized.");
     return NULL;
+  }
+    
```

Comment:
```suggestion
    Py_AtExit(cleanup);
```
Move the exit handler registration here.